datalad/downloaders/base.py

```diff
@@ -75,11 +75,12 @@ def __init__(self, credential=None, authenticator=None):
         authenticator: Authenticator, optional
           Authenticator to use for authentication.
         """
+        # Allow for anonymous connection if no credential is provided
```

Comment:
Is this comment misplaced?  The code immediately below is unchanged and doesn't check `credential`.  Perhaps it should be above the `if not credential` line below?

Reply:
I guess it was overall like a TODO comment reminder... I just removed it now